src/js/jsx/shared/Datalist.jsx

```diff
@@ -451,6 +475,15 @@ define(function (require, exports, module) {
             }
         },
 
+        removeAutofillSuggestion: function () {
+            if (this.props.useAutofill) {
+                this.setState({
+                    suggestTitle: ""
+                });
+                this.forceUpdate();
```

Comment:
I moved the call I was making in SCU to `inputChange`, which I think addresses both of the issues.